Bug #28853 causes failure in native build as build.properties is not on classpath

#28853 introduced a service that loads the build.properties file from the classloader.   When compiling native builds the builder removes everything it does not see as necessary.  Any resources that are needed at runtime need to be explicitly specified.   We have a file for this configuration in tools/dotcms-cli/cli/src/main/resources/resources-config.json

the build.properties file needs to be included in here

```
dotcli status
[ERROR] ❗  Error in command [status] with message:
  An exception NullPointerException Occurred With no error message provided.
run with -e or --errors for full details on the exception.
```

```
[ERROR] ❗  Error in command [login] with message:
  An exception NullPointerException Occurred With no error message provided.
java.lang.NullPointerException
	at java.base@11.0.18/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base@11.0.18/java.util.Optional.<init>(Optional.java:107)
	at java.base@11.0.18/java.util.Optional.of(Optional.java:120)
	at com.dotcms.api.BuiltVersionServiceImpl.version(BuiltVersionServiceImpl.java:38)
	at com.dotcms.api.BuiltVersionServiceImpl_ClientProxy.version(Unknown Source)
	at com.dotcms.api.provider.BuildVersionHeaders.update(BuildVersionHeaders.java:23)
	at com.dotcms.api.provider.BuildVersionHeaders_ClientProxy.update(Unknown Source)
	at org.jboss.resteasy.microprofile.client.header.ClientHeadersRequestFilter.lambda$filter$4(ClientHeadersRequestFilter.java:61)
	at java.base@11.0.18/java.util.Optional.ifPresent(Optional.java:183)
	at org.jboss.resteasy.microprofile.client.header.ClientHeadersRequestFilter.filter(ClientHeadersRequestFilter.java:61)
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.filterRequest(ClientInvocation.java:747)
	at org.jboss.resteasy.microprofile.client.impl.MpClientInvocation.filterRequest(MpClientInvocation.java:75)
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.invoke(ClientInvocation.java:492)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:152)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:115)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:76)
	at com.sun.proxy.$Proxy69.getToken(Unknown Source)
	at java.base@11.0.18/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.microprofile.client.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:148)
	at com.sun.proxy.$Proxy70.getToken(Unknown Source)
	at com.dotcms.api.DefaultAuthenticationContextImpl.login(DefaultAuthenticationContextImpl.java:117)
	at com.dotcms.api.DefaultAuthenticationContextImpl_ClientProxy.login(Unknown Source)
	at com.dotcms.cli.command.LoginCommand.call(LoginCommand.java:157)
	at com.dotcms.cli.command.LoginCommand_Subclass.call$$superforward1(Unknown Source)
	at com.dotcms.cli.command.LoginCommand_Subclass$$function$$6.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:129)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:33)
	at com.dotcms.cli.command.LoginCommand_Subclass.call(Unknown Source)

```

--- a/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/BuiltVersionServiceImpl.java
+++ b/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/BuiltVersionServiceImpl.java
@@ -35,7 +35,7 @@ public class BuiltVersionServiceImpl implements BuiltVersionService {
                 logger.error("Unable to load properties file", e);
             }
         }
-        return Optional.of(buildVersion);
+        return Optional.ofNullable(buildVersion);
 
     }
 

--- a/tools/dotcms-cli/api-data-model/src/main/resources/resources-config.json
+++ b/tools/dotcms-cli/api-data-model/src/main/resources/resources-config.json
@@ -10,7 +10,7 @@
       "pattern": ".*\\.dll$"
     },
     {
-      "pattern": ".*\\.properties$"
+      "pattern": "(.*\\/)?build\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/api-data-model/src/main/resources/resources-config.json
+++ b/tools/dotcms-cli/api-data-model/src/main/resources/resources-config.json
@@ -8,6 +8,9 @@
     },
     {
       "pattern": ".*\\.dll$"
+    },
+    {
+      "pattern": ".*\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/api-data-model/src/test/resources/resources-config.json
+++ b/tools/dotcms-cli/api-data-model/src/test/resources/resources-config.json
@@ -10,7 +10,7 @@
       "pattern": ".*\\.dll$"
     },
     {
-      "pattern": ".*\\.properties$"
+      "pattern": "(.*\\/)?build\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/api-data-model/src/test/resources/resources-config.json
+++ b/tools/dotcms-cli/api-data-model/src/test/resources/resources-config.json
@@ -8,6 +8,9 @@
     },
     {
       "pattern": ".*\\.dll$"
+    },
+    {
+      "pattern": ".*\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/cli/src/main/resources/resources-config.json
+++ b/tools/dotcms-cli/cli/src/main/resources/resources-config.json
@@ -10,7 +10,7 @@
       "pattern": ".*\\.dll$"
     },
     {
-      "pattern": ".*\\.properties$"
+      "pattern": "(.*\\/)?build\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/cli/src/main/resources/resources-config.json
+++ b/tools/dotcms-cli/cli/src/main/resources/resources-config.json
@@ -8,6 +8,9 @@
     },
     {
       "pattern": ".*\\.dll$"
+    },
+    {
+      "pattern": ".*\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/cli/src/test/resources/resources-config.json
+++ b/tools/dotcms-cli/cli/src/test/resources/resources-config.json
@@ -10,7 +10,7 @@
       "pattern": ".*\\.dll$"
     },
     {
-      "pattern": ".*\\.properties$"
+      "pattern": "(.*\\/)?build\\.properties$"
     }
   ]
 }

--- a/tools/dotcms-cli/cli/src/test/resources/resources-config.json
+++ b/tools/dotcms-cli/cli/src/test/resources/resources-config.json
@@ -8,6 +8,9 @@
     },
     {
       "pattern": ".*\\.dll$"
+    },
+    {
+      "pattern": ".*\\.properties$"
     }
   ]
 }

